osc-server/src/test/java/org/osc/core/broker/rest/client/k8s/KubernetesStatusApiTest.java

```diff
@@ -0,0 +1,167 @@
+
+    @Mock
+    private org.osc.core.broker.rest.client.k8s.KubernetesClient kubernetesClient;
+
+    @Mock
+    private DefaultKubernetesClient fabric8Client;
+
+    @Mock
+    private MixedOperation<ComponentStatus, ComponentStatusList, DoneableComponentStatus, Resource<ComponentStatus,DoneableComponentStatus>> mixedOperationMock;
+
+    private KubernetesStatusApi api;
+
+    @Before
+    public void testInitialize() throws Exception{
+        MockitoAnnotations.initMocks(this);
+        when(this.kubernetesClient.getClient()).thenReturn(this.fabric8Client);
+        this.api = new KubernetesStatusApi(this.kubernetesClient);
+
+        when(this.fabric8Client.componentstatuses()).thenReturn(this.mixedOperationMock);
+    }
+
+    @Test
+    public void testIsServiceReady_WhenK8sClientThrowsKubernetesClientException_ThrowsVmidcException() throws Exception {
+        // Arrange.
+        this.exception.expect(VmidcException.class);
+        when(this.fabric8Client.componentstatuses()).thenThrow(new KubernetesClientException(""));
+
+        // Act.
+        this.api.isServiceReady();
+    }
+
+    @Test
+    public void testIsServiceReady_WhenComponentStatusIsNull_ServiceNotReady() throws Exception {
+        // Arrange.
+        mockComponentStatus(null);
+
+        // Act.
+        boolean isServiceReady = this.api.isServiceReady();
+
+        // Assert.
+        assertFalse(isServiceReady);
+    }
+
+    @Test
+    public void testIsServiceReady_WhenComponentStatusConditionsIsNull_ServiceNotReady() throws Exception {
+        // Arrange.
+        ComponentStatus componentStatus = newComponentStatus(null, null);
+        componentStatus.setConditions(null);
+        mockComponentStatus(componentStatus);
+
+        // Act.
+        boolean isServiceReady = this.api.isServiceReady();
+
+        // Assert.
+        assertFalse(isServiceReady);
+    }
+
+
+    @Test
+    public void testIsServiceReady_WhenComponentStatusHealthyConditionNotFound_ServiceNotReady() throws Exception {
+        // Arrange.
+        ComponentStatus componentStatus = newComponentStatus("other_condition", "True");
+        mockComponentStatus(componentStatus);
+
+        // Act.
+        boolean isServiceReady = this.api.isServiceReady();
+
+        // Assert.
+        assertFalse(isServiceReady);
+    }
+
+
+    @Test
+    public void testIsServiceReady_WhenComponentStatusConditionFalse_ServiceNotReady() throws Exception {
+        // Arrange.
+        ComponentStatus componentStatus = newComponentStatus("Healthy", "False");
+        mockComponentStatus(componentStatus);
+
+        // Act.
+        boolean isServiceReady = this.api.isServiceReady();
+
+        // Assert.
+        assertFalse(isServiceReady);
+    }
+
+    @Test
+    public void testIsServiceReady_WhenComponentStatusConditionTrue_ServiceReady() throws Exception {
+        // Arrange.
+        ComponentStatus componentStatus = newComponentStatus("Healthy", "True");
```

Comment:
do you want to reference the constants defined in the KubernetesStatusApi? You can make them package private for this

Reply:
I wanted to actually use the strings. If the same string fields are referenced and they get changed accidentally (regression scenario) the tests wont flag that since the value is shared. If someone is intentionally changing the strings they must update the tests as well and the failure serves as a "are you sure you want to change this?" type of warning.

Reply:
makes sense. although unit tests should be testing the logic not the state though...but its fine. If these values change on k8s we will have to update in all the places...since its only 2 its less of a problem.